server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts

```diff
@@ -107,14 +120,43 @@ export function create(
 
             handleResponse(Promise.all([createP, tenantKeyP]).then(([_, key]) => {
                 // @TODO: Modify it to return an object only, it returns string for back-compat.
-                return generateToken
-                    ? {
+                if (enableDiscovery && generateToken) {
+                    return {
                         id,
+                        session,
                         token: getCreationToken(token, key, id),
-                    }
-                    : id;
+                    };
+                }
+                else if (enableDiscovery && !generateToken) {
```

Comment:
nit: please use the standard if..else if..else format in the repo
```ts
if (something) {
  // do something
} else if (otherthing) {
  // do something else
} else {
  // do thing
}
```

Reply:
Got it. Done.